frontend/src/features/admin-form/create/builder-and-design/utils/fieldCreation.ts

```diff
@@ -160,3 +175,87 @@ export const getFieldCreationMeta = (fieldType: BasicField): FieldCreateDto => {
     }
   }
 }
+
+export const getMyInfoFieldCreationMeta = (
+  myInfoAttribute: MyInfoImplementedTypes,
+): MyInfoField => {
+  const baseMeta: Pick<
+    MyInfoField,
+    'disabled' | 'required' | 'title' | 'description' | 'fieldType' | 'myInfo'
+  > = {
+    disabled: false,
+    required: true,
+    title: MYINFO_FIELD_TO_DRAWER_META[myInfoAttribute].label,
+    description: '',
+    fieldType: MYINFO_FIELD_CONSTANTS[myInfoAttribute].fieldType,
+    myInfo: {
+      attr: myInfoAttribute,
+    },
+  }
+
+  switch (myInfoAttribute) {
+    case MyInfoAttribute.Name:
+    case MyInfoAttribute.PassportNumber:
+    case MyInfoAttribute.VehicleNo:
+    case MyInfoAttribute.RegisteredAddress:
+    case MyInfoAttribute.Employment: {
+      return {
+        ...baseMeta,
+        fieldType: BasicField.ShortText,
+        ...myInfoTextFieldMeta,
+      }
+    }
+
+    case MyInfoAttribute.DateOfBirth:
+    case MyInfoAttribute.PassportExpiryDate:
+    case MyInfoAttribute.WorkpassExpiryDate:
+    case MyInfoAttribute.MarriageDate:
+    case MyInfoAttribute.DivorceDate: {
+      return {
+        ...baseMeta,
+        fieldType: BasicField.Date,
+        ...myInfoDateFieldMeta,
+      }
+    }
+
+    case MyInfoAttribute.Sex:
+    case MyInfoAttribute.Race:
+    case MyInfoAttribute.Nationality:
+    case MyInfoAttribute.BirthCountry:
+    case MyInfoAttribute.ResidentialStatus:
+    case MyInfoAttribute.Dialect:
+    case MyInfoAttribute.HousingType:
+    case MyInfoAttribute.HdbType:
+    case MyInfoAttribute.Occupation:
+    case MyInfoAttribute.WorkpassStatus:
+    case MyInfoAttribute.Marital:
+    case MyInfoAttribute.CountryOfMarriage: {
+      return {
+        ...baseMeta,
+        fieldType: BasicField.Dropdown,
+        ...getMyInfoDropdownFieldMeta(myInfoAttribute),
+      }
+    }
+
+    case MyInfoAttribute.MarriageCertNo: {
+      return {
+        ...baseMeta,
+        fieldType: BasicField.Number,
```

Comment:
i think this should still be of shorttext type, doesn't make sense for the number to be incrementable or decrementable haha

Reply:
even if they are number-only, makes sense to be used as shorttext regardless just in case haha.